### PR TITLE
Fixed a bug in ArchiveExtractor that caused it to fail on case-insens…

### PR DESCRIPTION
**Pull request by MikeThomsen**
…itive file systems.

**Summary**

**Tests and Documentation**

### CODE REVIEW

**Comment by eirslett** [comment]
Thanks! Looks like this patch is needed by more people (#813) but the tests fail on Travis... could you see if you can figure out the reason why, and make them pass somehow?

**Comment by MikeThomsen** [comment]
@eirslett I'll try to find some time soon to figure out what's going on.

**Comment by MikeThomsen** [comment]
It was an `assert` in `yarn-integration/verify.groovy` that was causing it. The relevant line it was looking for no longer appears at all in the build log. Left a reminder TODO for you to see if that part of the test needs something new.

File: frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java [hunk 1]
```diff
@@ -139,4 +141,46 @@ public void extract(String archive, String destinationDirectory) throws ArchiveE
...
+            /*
+             * The first check should handle case-sensitive file systems. We need this
+             * in order to weed out case-sensitive file systems with a non-existent path
+             * that slipped through the first test.
+             */
+            if (new File(destPath).exists() && !(new File(destPath.toLowerCase()).exists())) {
+                return false;
+            }
+
+            boolean retVal = true;
```
Comment (eirslett) [inline frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java]: Why is this stuff going on? Isn't it easier with `destPath.toLowerCase().startsWith(destDir.toLowerCase())`?

File: frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java [hunk 2]
```diff
@@ -139,4 +141,46 @@ public void extract(String archive, String destinationDirectory) throws ArchiveE
...
+    /**
+     * Do multiple file system checks that should enable the plugin to work on any file system
+     * whether or not it's case sensitive or not.
+     *
+     * @param destPath
+     * @param destDir
+     * @return
+     */
+    private boolean startsWithPath(String destPath, String destDir) {
+        if (destDir.startsWith(destDir)) {
```
Comment (eirslett) [inline frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java]: Won't this check always be true? Is it supposed to be `destPath`?

Comment (MikeThomsen) [reply frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java]: That was a typo on my part.

File: frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java [hunk 3]
```diff
@@ -139,4 +141,26 @@ public void extract(String archive, String destinationDirectory) throws ArchiveE
...
+     * @param destDir
+     * @return
+     */
+    private boolean startsWithPath(String destPath, String destDir) {
+        if (destPath.startsWith(destDir)) {
+            return true;
+        } else if (destDir.length() > destPath.length()) {
+            return false;
+        } else {
+            if (new File(destPath).exists() && !(new File(destPath.toLowerCase()).exists())) {
```
Comment (eirslett) [inline frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java]: I don't understand why this check is here, why is it checking whether a file exists?

Comment (MikeThomsen) [reply frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java]: The path that Maven was feeding your plugin was not the same case as what was on the file system. However, HPFS+ and NTFS are case insensitive by default so the folder does actually exist and technically that exception should not be thrown.

FWIW, we fixed **our** issue by finally stomping the inconsistencies on a Linux box IIRC. Based on the other users' comments, I'm not entirely sure their issues are the same so you might want to just be aware of this particular scenario and ask them for more details.

Comment (MikeThomsen) [reply frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java]: That said, I think all of the other users were running Windows or macOS.

**Comment by eirslett** [comment]
ok, thank you!

**Comment by eirslett** [comment]
There we go, 1.9.0 is released, merry christmas! 🎅 ❄️ 🎄

**Pull request merged**